src/utils/amm.ts

```diff
@@ -72,10 +73,13 @@ export const prepareTransactionForAMM = async (
     parsedTotal: BigNumber,
     parsedSlippage: BigNumber,
     collateral: string | undefined,
-    referral: string | null
+    referral: string | null,
+    networkId: Network
 ): Promise<ethers.ContractTransaction> => {
     let tx: ethers.ContractTransaction;
 
+    const gasLimit = networkId === 10 ? 3000000 : null; // set gasLimit to 3M on Optimism
```

Comment:
Do not use hardcoded values for network (10) and gasLimit (3000000). Set constant.

Reply:
Done!